src/stdune/list.mli

```diff
@@ -43,3 +43,5 @@ val assoc : ('a * 'b) t -> 'a -> 'b option
 val singleton : 'a -> 'a t
 
 val nth : 'a t -> int -> 'a option
+
+val physically_eq : 'a t -> 'a t -> bool
```

Comment:
what is the rationale for this specialized function? Polymorphic physical equality seems fine to me

Reply:
That's to avoid cases when there's `a == b` somewhere with `a` and `b` lists, but for some reason the type of `a` and `b` change, and physical equality might not be what we're after. In that case, this becomes a compile error to remember us to check.

Reply:
I see. I guess I'm just not excited at the idea of seeing this function pop-up in every module, but if it can help catch problems in the future why not. The name looks a bit odd to me though, what about `physically_equal`?

Reply:
That's just useful for modules for which we rely on physical equality at some point, so probably only data structures.

Good point with the name, I'll do this.